datafusion/physical-plan/src/sorts/cursor.rs

```diff
@@ -275,6 +324,26 @@ impl<T: FieldValues> Cursor for FieldCursor<T> {
         self.offset += 1;
         t
     }
+
+    fn slice(&self, offset: usize, length: usize) -> Result<Self> {
+        let FieldCursor {
+            values,
+            offset: _,
+            null_threshold,
+            options,
+        } = self;
+
+        Ok(Self {
+            values: values.slice(offset, length)?,
+            offset: 0,
+            null_threshold: *null_threshold,
```

Comment:
Is this correct?

Reply:
You are correct. In order to recalculate the null_threshold, we rely on methods from the original `FieldArray: Array` interface (not the inner `FieldValues`). Is there any reason not to keep the original field array?